ipctest/src/main.rs

```diff
@@ -27,14 +27,34 @@ use crate::errors::*;
 // Run with 'RUST_LOG=run,audioipc cargo run -p ipctest'
 #[cfg(unix)]
 fn run() -> Result<()> {
+    use std::ffi::CString;
+
     let handle =
         unsafe { audioipc_server::audioipc_server_start(std::ptr::null(), std::ptr::null()) };
     let fd = audioipc_server::audioipc_server_new_client(handle);
+    let fd = unsafe {
+        let new_fd = libc::dup(fd);
+        libc::close(fd);
+        new_fd
+    };
+    assert!(fd >= 0);
+
+    let args: Vec<String> = std::env::args().collect();
 
     match unsafe { libc::fork() } {
         -1 => bail!("fork() failed"),
         0 => {
-            return client::client_test(fd);
+            let self_path = CString::new(&*args[0]).unwrap();
+            let child_arg1 = CString::new("--client").unwrap();
+            let child_arg2 = CString::new("--fd").unwrap();
+            let child_arg3 = CString::new(format!("{}", fd)).unwrap();
+            let child_args = [self_path.as_c_str().as_ptr(),
```

Comment:
`self_path.as_ptr()` could work. It seems [`CString::as_ptr()`](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.as_ptr) is a [shortcut for `CString::as_c_str()::as_ptr()`](https://doc.rust-lang.org/std/ffi/struct.CString.html#extracting-a-raw-pointer-to-the-whole-c-string).

Reply:
That's tidier, thanks!